File: skthackaton/src/components/pages/FAQ.js
import React, { useState, useEffect } from 'react';
import axios from "axios";
import '../../App.css';
import Footer from '../Footer';

function FAQ() {
  const [activeTab, setActiveTab] = useState(null);
//   const [faqInfo, setFaqInfo] = useState([]); // 이미지 정보 상태

  const toggleTab = (index) => {
    if (activeTab === index) {
      setActiveTab(null);
    } else {
      setActiveTab(index);
    }
  };

//   useEffect(() => {
//     axios
//       .post(`http://49.50.162.196:8000/aidoctor/`)
//       .then((response) => {
//         // 서버에서 받은 응답 데이터
//         const serverFaqInfo = response.data;

//         // 서버에서 받아온 이미지 정보를 현재 이미지 정보에 추가
//         const updatedFaqInfo = serverFaqInfo.map((item) => ({
//             question: item.picture_url,
//             answer: item.word,
//         }));

//         // imageInfo 배열 업데이트
//         setFaqInfo(updatedFaqInfo);
//       })
//       .catch((error) => {
//         // 오류 발생 시의 처리
//       })
//   }, []); // 빈 배열을 전달하여 한 번만 호출되도록 설정

  const faqInfo = [
    {
      question: 'When will my order arrive?',
      answer: 'Shipping time is set by our delivery partners, according to the delivery method chosen by you. Additional details can be found in the order confirmation.',
    },
    {
      question: 'How do I track my order?',
      answer: 'Once shipped, you’ll get a confirmation email that includes a tracking number and additional information regarding tracking your order.',
    },
    {
      question: 'What’s your return policy?',
      answer: 'We allow the return of all items within 30 days of your original order’s date. If you’re interested in returning your items, send us an email with your order number and we’ll ship a return label.',
    },
    {
      question: 'How do I make changes to an existing order?',
      answer: 'Changes to an existing order can be made as long as the order is still in “processing” status. Please contact our team via email and we’ll make sure to apply the needed changes. If your order has already been shipped, we cannot apply any changes to it. If you are unhappy with your order when it arrives, please contact us for any changes you may require.',
    },
    {
      question: 'What shipping options do you have?',
      answer: 'For USA domestic orders we offer FedEx and USPS shipping.',
    },
    {
      question: 'What payment methods do you accept?',
      answer: 'Any method of payments acceptable by you. For example: We accept MasterCard, Visa, American Express, PayPal, JCB Discover, Gift Cards, etc.',
    },
  ];

  return (
    <>
    <div className="App-back">
    <div className="flex justify-center items-start my-2">
      <div className="w-full sm:w-10/12 md:w-1/2 my-1">
        <br></br><br></br><br></br>
        <h2 className="text-xl font-semibold text-vnet-blue mb-2">FAQ</h2><br></br>
        <ul className="flex flex-col">
          {faqInfo.map((faq, index) => (
            <li className="bg-white my-2 shadow-lg" key={index}>
              <h2
                onClick={() => toggleTab(index)}
                className="flex flex-row justify-between items-center font-semibold p-3 cursor-pointer"
              >
                <span>{faq.question}</span>
                <svg
                  className={`fill-current text-blue-700 h-6 w-6 transform transition-transform duration-500 ${activeTab === index ? 'rotate-180' : ''}`}
                  viewBox="0 0 20 20"
                >
                  <path d="M13.962,8.885l-3.736,3.739c-0.086,0.086-0.201,0.13-0.314,0.13S9.686,12.71,9.6,12.624l-3.562-3.56C5.863,8.892,5.863,8.611,6.036,8.438c0.175-0.173,0.454-0.173,0.626,0l3.25,3.247l3.426-3.424c0.173-0.172,0.451-0.172,0.624,0C14.137,8.434,14.137,8.712,13.962,8.885 M18.406,10c0,4.644-3.763,8.406-8.406,8.406S1.594,14.644,1.594,10S5.356,1.594,10,1.594S18.406,5.356,18.406,10 M17.521,10c0-4.148-3.373-7.521-7.521-7.521c-4.148,0-7.521,3.374-7.521,7.521c0,4.147,3.374,7.521,7.521,7.521C14.148,17.521,17.521,14.147,17.521,10"></path>
                </svg>
              </h2>
              <div
                className="border-l-2 border-blue-600 overflow-hidden max-h-0 duration-500 transition-all"
                style={{
                  maxHeight: activeTab === index ? `${faq.answer.length * 20}px` : '0px',
                }}
              >
                <p className="p-3 text-gray-900">
                  {faq.answer}
                </p>
              </div>
            </li>
          ))}
        </ul>
      </div>
    </div>
    </div>
    <Footer />
    </>
  );
}

export default FAQ;

File: faq.py
def get_faq():
    a = "콜라주 예시를 똑같이 따라해야하나요?"
    b = "아닙니다. 자유롭게 배치가능합니다"
    c = "사용 과정에서 아이와의 의사소통을 할때 권장하는 행동이 있을까요?"
    d = "아동이 시도하는 모든 말을 반복하여 따라말해주신다면 아이의 소통 능력 향상에 도움이 됩니다."
    e = "감정 선택시 랜덤으로 보호자가 선택해도 되나요?"
    f = "네, 감정 선택 시 보호자가 랜덤으로 선택해도 괜찮습니다."
    g = "청각이 예민한 자녀라 음악이 안나오게 하고 싶은데 어떻게 해야하나요?"
    h = "서비스 사용 전에 소리를 끄고 사용해주시길 바랍니다."

    qa = [{"question":a, "answer":b}, {"question":c, "answer":d},{"question":e,"answer":f},{"question":g,"answer":h}]

    i = "게시판에 게시할때 콜라주 결과 이외의 사진을 올려도 되나요? "
    j = "물론입니다.서비스 활용 전반에 관련된 사진은 모두 업로드 가능합니다."
    k = "회원 가입을 했는데 뉴스레터가 안와요."
    l = "뉴스레터는 홈페이지 하단에 이메일란에 주소를 입력해주시면 전송됩니다."
   
    qa = {"data":[{"qustion":a, "answer":b}, {"question":c, "answer":d},{"question":e,"answer":f},{"question":g,"answer":h},{"question":i, "answer":j},{"question":k,"answer":l}]}


    print(qa)
    return qa